### PR TITLE
Revert "Revert count(where:)"

### PR DESCRIPTION
**Pull request by natecook1000**
This reverts commit 779ea19a6a599a62ac6c082e6cee870fadc7f675, to implement the re-acceptance of SE-0220.

https://github.com/apple/swift-evolution/blob/main/proposals/0220-count-where.md

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/SequenceAlgorithms.swift [hunk 1]
```diff
@@ -572,6 +572,50 @@ extension Sequence where Element: Equatable {
...
+  ///     // duckCount == 4
+  ///
+  /// The sequence must be finite.
+  ///
+  /// - Parameter predicate: A closure that takes each element of the sequence
+  ///   as its argument and returns a Boolean value indicating whether
+  ///   the element should be included in the count.
+  /// - Returns: The number of elements in the sequence that satisfy the given
+  ///   predicate.
+  @inlinable
```
Comment (harlanhaskins) [inline stdlib/public/core/SequenceAlgorithms.swift:605]: Does this need availability? Could we make this `@backDeploy`ed?

Comment (Azoy) [reply stdlib/public/core/SequenceAlgorithms.swift:605]: The standard library can't effectively use `@backDeploy` yet as there are, issues... I would however throw an `@_alwaysEmitIntoClient` on.

Comment (natecook1000) [reply stdlib/public/core/SequenceAlgorithms.swift]: @benrimmington We're handling the addition of typed throws throughout the standard library separately. The loop is fine here.

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by xwu** [comment]
@swift-ci please smoke test

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by natecook1000** [comment]
@swift-ci Please test

File: stdlib/public/core/SequenceAlgorithms.swift [hunk 2]
```diff
@@ -572,6 +572,48 @@ extension Sequence where Element: Equatable {
...
+  ///     // duckCount == 4
+  ///
+  /// The sequence must be finite.
+  ///
+  /// - Parameter predicate: A closure that takes each element of the sequence
+  ///   as its argument and returns a Boolean value indicating whether
+  ///   the element should be included in the count.
+  /// - Returns: The number of elements in the sequence that satisfy the given
+  ///   predicate.
+  @_alwaysEmitIntoClient
```
Comment (phausler) [inline stdlib/public/core/SequenceAlgorithms.swift:605]: does this not need the ABI shims since it is AEIC?

Comment (Azoy) [reply stdlib/public/core/SequenceAlgorithms.swift:605]: ?

Comment (natecook1000) [reply stdlib/public/core/SequenceAlgorithms.swift:605]: This is brand-new API for the standard library, and it needs to be AEIC due to typed throws, so there's no existing ABI to worry about 👍🏻

**Review by phausler — APPROVED** [review]

**Review by glessard — APPROVED** [review]

**Pull request merged**